glommio/src/channels/shared_channel.rs

```diff
@@ -524,8 +524,8 @@ mod test {
             })
             .unwrap();
 
-        ex1.join().unwrap();
-        ex2.join().unwrap();
+        ex1.join().unwrap().unwrap();
```

Comment:
At some point it might make more sense to have the tests return Result<(), E> type thing so we can maybe get away with using `?` in a bunch of these unwraps.. obviously I haven't worked this idea full through, but it _might_ produce a nicer result and an easier pattern than unwrapping so many unwraps etc.. Thoughts?

Reply:
Agreed, these chains of `unwrap` are not nice. I am thinking of wrapping the `JoinHandle` into one of our making to flatten the two errors (from creating the thread and from creating the local executor inside the thread) into a single one.

Reply:
Yes! Love the flattening idea. I think that would clean up a lot of stuff and make the api cleaner.